packages/kit/src/pages.ts

```diff
@@ -34,3 +34,32 @@ export function extendRouteRules (route: string, rule: NitroRouteConfig, options
       : defu(opts.routeRules[route], rule)
   }
 }
+
+export interface AddRouteMiddlewareOptions {
+  /**
+   * Override middleware path
+   *
+   * @default false
+   */
+  override?: boolean
+}
+
+export function addRouteMiddleware (input: NuxtMiddleware | NuxtMiddleware[], options: AddRouteMiddlewareOptions = {}) {
+  const nuxt = useNuxt()
+  nuxt.hook('app:resolve', (app) => {
+    const middlewares = Array.isArray(input) ? input : [input]
+
+    middlewares.forEach((middleware) => {
```

Comment:
In general we prefer for...of rather than forEach in Nuxt.

```suggestion
    for (const middleware of middlewares) {
```